src/managers/config.py

```diff
@@ -582,4 +439,4 @@ def set_environment(self) -> None:
             self.jvm_performance_opts,
             self.heap_opts,
         ]
-        update_env(env=map_env(env=updated_env_list))
+        self.workload.update_environment(env=self.workload.map_env(env=updated_env_list))
```

Comment:
**todo:** Given that we only call `update_environment` here, we might as well just put the logic of that function in this `set_environment`, and remove it from workload. What do you think?

Reply:
I don't know tbh. `update_environment` also calls `get_env` and `map_env`, which are helpers, so we would end up with `config.py -> set_environmet()` having like 70 lines of code. I guess it's fine since we won't need to call it anywhere else in the future?

Reply:
Isn't it just 1 small function?

```python
    def set_environment(self, env: dict[str, str]) -> None:
        """Updates the /etc/environment for the workload.

        Args:
            env: dict of key env-var, value
        """
        map_env = {}
        for var in self.current_env:
            key = "".join(var.split("=", maxsplit=1)[0])
            value = "".join(var.split("=", maxsplit=1)[1:])
            if key:
                # only check for keys, as we can have an empty value for a variable
                map_env[key] = value

        updated_env = map_env | env
        content = "\n".join([f"{key}={value}" for key, value in updated_env.items()])

        self.workload.write(content=content, path="/etc/environment")

```

Reply:
Sort of, the map_env loop is used on the current_env as well. Take a look now.